pkg/cluster/controllers/storage/minio.go

```diff
@@ -184,6 +174,71 @@ func (m *MinIOController) getTenantsServiceName(harborcluster *goharborv1.Harbor
 	return defaultMinIOService
 }
 
+const (
+	bucketInitializatedAnnotationKey = "minio.harbor.goharbor.io/bucket-initialized"
+)
+
+func (m *MinIOController) initializBucket(ctx context.Context, harborcluster *goharborv1.HarborCluster, tenant *miniov2.Tenant) (*lcm.CRStatus, error) {
+	if m.isBucketInitialized(ctx, tenant) {
+		return nil, nil
+	}
+
+	// Apply minio init job
+	if crs, err := m.applyMinIOInitJob(ctx, harborcluster); err != nil {
+		return crs, err
+	}
+
+	job, ready, err := m.checkMinIOInitJobReady(ctx, harborcluster)
+	if err != nil {
+		return minioNotReadyStatus(GetInitJobError, err.Error()), err
+	}
+
+	if !ready {
+		m.Log.Info("MinIO init job is not ready yet")
+
+		return minioUnknownStatus(), nil
+	}
+
+	if err := m.setBucketInitialized(ctx, tenant); err != nil {
+		return minioNotReadyStatus(UpdateMinIOError, err.Error()), err
+	}
+
+	if err := m.KubeClient.Delete(ctx, job, client.PropagationPolicy(metav1.DeletePropagationBackground)); err != nil {
+		return minioNotReadyStatus(DeleteInitJobError, err.Error()), err
+	}
+
+	return nil, nil
+}
+
+func (m *MinIOController) isBucketInitialized(_ context.Context, tenant *miniov2.Tenant) bool {
+	annotations := tenant.GetAnnotations()
+	if annotations == nil {
+		return false
+	}
+
+	s, ok := annotations[bucketInitializatedAnnotationKey]
+	if !ok {
+		return false
+	}
+
+	v, _ := strconv.ParseBool(s)
```

Comment:
```suggestion
	if v, err := strconv.ParseBool(s); err != nil {
		logger.Get(ctx).Error(err, "unexpected value for annotation", "annotation", bucketInitializatedAnnotationKey, "value", s) // Not sure about the syntax :s
		
		return false // Explicitly return false in case of error
	}
```